node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -408,9 +410,23 @@ removeNextUpdate = wsReadyUpdates %= drop 1
 testReset :: Update ()
 testReset = put def
 
-updateHistoryCache :: CId Wal -> [TxHistoryEntry] -> Update ()
-updateHistoryCache cWalId cTxs =
-    wsHistoryCache . at cWalId ?= cTxs
+data HistoryCacheUpdate
+    = InitHistoryCache
+    | InsertToHistoryCache [TxHistoryEntry]
+    | RemoveFromHistoryCache [TxHistoryEntry]
+
+updateHistoryCache :: CId Wal -> HistoryCacheUpdate -> Update ()
```

Comment:
We cannot change the type of existing transaction actions. This would break DB tx log replay. We can introduce new transactions but not change or remove existing ones.

Reply:
Since Philipp and I are changing this `updateHistoryCache` too, I suggest we rebase this PR on that, and I think it'd also make sense to just covert this into 3 new transactions, for clear, insert and delete.